pkg/csi/service/wcpguest/controller.go

```diff
@@ -332,12 +335,30 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 				c.supervisorNamespace, err)
 			log.Error(msg)
 			eventList, err := c.supervisorClient.CoreV1().Events(c.supervisorNamespace).List(ctx,
-				metav1.ListOptions{FieldSelector: "involvedObject.name=" + pvc.Name})
+				metav1.ListOptions{
+					FieldSelector:        "involvedObject.name=" + pvc.Name,
+					ResourceVersion:      pvc.ResourceVersion,
```

Comment:
Is it guaranteed that the resource version of the pvc in the above Create() call is same as the one which receives the Provisioning Failed error ?

Reply:
If you look at the last parameter to `ListOptions`,
`ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan`

This means we are not looking for an exact match but equal or greater resource version than the one we pass to `isPVCInSupervisorClusterBound` method.

Reply:
> what if we observe ConflictError from k8s saying "ResourceVersion is already modified" ?
When we list the events on the supervisor namespace for the given PVC, we are not modifying the PVC. 

The `ResourceVersion` we pass via the options to this List() call is the one we have received in a previous Get() call. 
The `ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan` is set to `NotOlderThan`. NotOlderThan matches data at least as new as the provided `ResourceVersion`. 

The PVC ResourceVersion at the time of the List() call might be higher and different but we are not querying for that exact version. 

Considering this, do you think we can get this error?

Reply:
No need to handle this error. I misread it as a Get() call. You can ignore my comment